src/azure-cli/azure/cli/__main__.py

```diff
@@ -115,6 +122,9 @@ def cli_main(cli, args):
     except Exception as ex:  # pylint: disable=broad-except
         logger.debug("Intercept survey prompt failed. %s", str(ex))
 
+    if 'AZURE_CLI_DISABLE_AZURELINUX2_WARNING' not in os.environ:
+        logger.warning(AZURELINUX2_WARNING_MESSAGE)
```

Comment:
The warning can also be disabled by `az config set core.only_show_errors=true`, but as https://github.com/Azure/azure-cli/pull/29628#discussion_r1701144179 mentioned, this will disable other warning messages too.

Introducing an environment variable `AZURE_CLI_DISABLE_AZURELINUX2_WARNING` will make it possible to disable only this warning message.